Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java

```diff
@@ -228,10 +312,23 @@ private boolean isCloseOnSingleTapDay() {
 
     public void onClick(View view) {
         switch (view.getId()) {
-            case R.id.city:
-                //TODO :: show a dialog with list of cities
+            case R.id.select_city:
+                new HotelSearchDialogCompat(HotelsActivity.this, getString(R.string.search_title),
+                        getString(R.string.search_hint), null, mSearchCities,
+                        new SearchResultListener<HotelSearchModel>() {
+                            @Override
+                            public void onSelected(BaseSearchDialogCompat dialog,
+                                                   HotelSearchModel item, int position) {
+                                String selectedCity = item.getId();
+                                String title = getString(R.string.showing) + " " + item.getName()
```

Comment:
You can make use of placeholder string for making this : https://stackoverflow.com/questions/3656371/dynamic-string-using-string-xml